modules/renter/contractor/update.go

```diff
@@ -58,11 +86,11 @@ func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
 	}
 	c.mu.Unlock()
 
-	// Only attempt contract formation/renewal if we are synced
-	// (harmless if not synced, since hosts will reject our renewal attempts,
-	// but very slow).
+	// Perform contract maintenance if our blockchain is synced. Use a separate
+	// goroutine so that the rest of the contractor is not blocked during
+	// maintenance.
 	if cc.Synced {
-		// Perform the contract maintenance in a separate thread.
 		go c.threadedContractMaintenance()
```

Comment:
Does this really need to happen every ~10 minutes?

Reply:
Potentially not, given that hosts tend to be in flux over a period of days. On the other hand, the process does not take very many computational resources to complete, I imagine just a few milliseconds. We don't do it while syncing, and we do it in a goroutine which means it won't affect other modules in the chain.

Overall, it didn't seem harmful to do it frequently, and doing it based on the consensus pulse means we don't need a background thread.